internal/session/command.go

```diff
@@ -14,39 +12,29 @@ import (
 type command struct {
 	tag string
 	cmd *proto.Command
+	err error
 }
 
-func (s *Session) getCommandCh(ctx context.Context, del string) <-chan command {
+func (s *Session) startCommandReader(ctx context.Context, del string) <-chan command {
 	cmdCh := make(chan command)
 
 	go func() {
 		labels := pprof.Labels("go", "CommandReader", "SessionID", strconv.Itoa(s.sessionID))
 		pprof.Do(ctx, labels, func(_ context.Context) {
 			defer close(cmdCh)
+
 			for {
 				tag, cmd, err := s.readCommand(del)
-				if err != nil {
-					if errors.Is(err, io.EOF) {
-						return
-					} else if err := response.Bad(tag).WithError(err).Send(s); err != nil {
+
+				select {
+				case cmdCh <- command{tag: tag, cmd: cmd, err: err}:
+					if err != nil {
```

Comment:
should continue reading future commands here rather than erroring out

Reply:
Fixed.